pennylane/templates/state_preparations/qrom_state_prep.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the QROMStatePreparation template.
+"""
+
+import itertools
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import Operation
+from pennylane.wires import Wires
+
+
+def _x_to_binary(n_precision, x):
+    r"""Converts a value within the range [0, 1) to its binary representation with a specified precision.
+
+    Args:
+        n_precision (int): The number of bits to use for the binary representation.
+        x (float): The value to convert to binary. Must be in the range [0, 1).
+
+    Returns:
+        str: The binary representation of the value, with the specified precision.
+
+    Example:
+        >>> _x_to_binary(3, 0.5)
+        '100'
+
+        Expected value as the binary representation of `0.5` is `0.100`.
+    """
+
+    binary_rep = bin(int(2 ** (n_precision + 1) + 2 ** (n_precision) * x))
+    if binary_rep[-n_precision - 1] == "1":
+        return "1" * n_precision
+
+    return binary_rep[-n_precision:]
```

Comment:
Very elegant! I think we should actually add this to the math module ! Maybe in another PR.